test/autorest/generated/paginggroup/pollers.go

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+// Changes may cause incorrect behavior and will be lost if the code is regenerated.
+
+package paginggroup
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+// PagingGetMultiplePagesLroPoller provides polling facilities until the operation completes
+type PagingGetMultiplePagesLroPoller interface {
+	Poll(context.Context) bool
+	Response() (ProductResultPager, error)
+	ResumeToken() (string, error)
+	Wait(ctx context.Context, pollingInterval time.Duration) (ProductResultPager, error)
+}
+
+type pagingGetMultiplePagesLroPoller struct {
+	// the client for making the request
+	client *pagingOperations
+	// polling tracker
+	pt pollingTracker
+}
+
+// Poll returns false if there was an error or polling has reached a terminal state
+func (p *pagingGetMultiplePagesLroPoller) Poll(ctx context.Context) bool {
+	done, err := p.done(ctx)
+	if err != nil {
+		return false
+	}
+	return !done
+}
+
+// Response returns the latest response that is stored from the latest polling operation
+func (p *pagingGetMultiplePagesLroPoller) Response() (ProductResultPager, error) {
+	client := p.client
+	u, err := url.Parse(p.pt.pollingURL())
+	if err != nil {
+		return nil, err
+	}
+	req := azcore.NewRequest(http.MethodGet, *u)
```

Comment:
I'm surprised we have to make another request here, i.e. when the poller reaches a successful terminal state I would expect we'd already have the first page of data.  I realize though that this is incongruent with how pagers work in the non-LRO case.  In addition, the polling URL isn't guaranteed to return the result, it's service-dependent (the "final GET" URL).